.github/workflows/reusable-build-and-push.yml

```diff
@@ -39,112 +52,88 @@ jobs:
+            suffix=
+          fi
 
-  vmclarity-ui:
-    needs:
-      - images
-      - timestamp
-    uses: ./.github/workflows/build-and-push-component.yaml
-    with:
-      dockerfile: Dockerfile.ui
-      image_name: ${{ needs.images.outputs.ui-image }}
-      image_tag: ${{ inputs.image_tag }}
-      push: ${{ inputs.push }}
-      timestamp: ${{ needs.timestamp.outputs.timestamp }}
+          echo "suffix=${suffix}" >> "$GITHUB_OUTPUT"
 
-  vmclarity-cli:
-    needs:
-      - images
-      - timestamp
-    uses: ./.github/workflows/build-and-push-component.yaml
-    with:
-      dockerfile: Dockerfile.cli
-      image_name: ${{ needs.images.outputs.cli-image }}
-      image_tag: ${{ inputs.image_tag }}
-      push: ${{ inputs.push }}
-      timestamp: ${{ needs.timestamp.outputs.timestamp }}
+      - name: List targets
+        id: targets
+        uses: docker/bake-action/subaction/list-targets@v4
+        with:
+          target: ${{ inputs.bake-group }}
 
-  vmclarity-cr-discovery-server:
+  build-and-push:
     needs:
-      - images
-      - timestamp
+      - prepare
+    strategy:
+      fail-fast: false
+      matrix:
+        target: ${{ fromJson(needs.prepare.outputs.targets) }}
     uses: ./.github/workflows/build-and-push-component.yaml
     with:
-      dockerfile: Dockerfile.cr-discovery-server
-      image_name: ${{ needs.images.outputs.cr-discovery-server-image }}
+      image_name: "${{ needs.prepare.outputs.registry }}/${{ matrix.target }}${{ needs.prepare.outputs.suffix }}"
       image_tag: ${{ inputs.image_tag }}
       push: ${{ inputs.push }}
-      timestamp: ${{ needs.timestamp.outputs.timestamp }}
+      timestamp: ${{ needs.prepare.outputs.timestamp }}
+      bake_target_name: ${{ matrix.target }}
+      post_image_tags: ${{ inputs.post_image_tags }}
 
   post-images:
     if: github.event_name == 'pull_request' && inputs.post_image_tags
     runs-on: ubuntu-latest
     needs:
-      - images
-      - vmclarity-apiserver
-      - vmclarity-orchestrator
-      - vmclarity-ui-backend
-      - vmclarity-ui
-      - vmclarity-cli
-      - vmclarity-cr-discovery-server
+      - prepare
+      - build-and-push
     steps:
+      - name: Download image names
+        uses: actions/download-artifact@v4
+        with:
+          pattern: ${{ github.run_id }}-*
+          merge-multiple: true
+          path: /tmp/images
+
+      - name: Get image names
+        run: |
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          {
+            echo "images<<$EOF"; cat /tmp/images/*; echo "$EOF"
+          } >> "$GITHUB_ENV"
+
       - name: Post comment with image tags
         uses: marocchino/sticky-pull-request-comment@331f8f5b4215f0445d3c07b4967662a32a2d3e31 # v2
         with:
           hide_and_recreate: true
           hide_classify: "OUTDATED"
           skip_unchanged: true
           header: image-tags
+          append: true
```

Comment:
what if we used `recreate: true` instead of adding new ones to ensure that only the latest built images are present instead of showing the comment for every commit?

Reply:
Oh, I must have left this line here by accident btw, thanks for noticing!